### ISSUE TITLE
where D_class use?

### ISSUE THREAD

**Issue by shenjun1994** [toandaominh1997/EfficientDet.Pytorch#118]
you  have defined D_class in efficientdet，but i don't find you have used it in net?

**Comment by rmcavoy** [comment]
From examining the original paper and this code, I believe D_class needs to be set by setting the optional parameter stacked_convs as an argument to the retinahead. To be blunt, I would would recommend basing your work off a different code as there are some massive bugs in this code that the repo owner seems to have no interest in fixing like the fact that it randomly reinitalizes the pretrained efficientnet backbone directly after loading (which should prevent it from stably training at all), this critical D_class bug, and the multiple minor issues/bugs that people pointed out with math/implementation of the bifpn, retinahead and loss.

**Comment by macwilam** [comment]
@rmcavoy do you maybe know any other code for efficientdet publicly available that you consider good? Any recommendations?

https://github.com/signatrix/efficientdet seems to only support 0 size. It is similar for most other pytorch  implementations.

**Comment by rmcavoy** [comment]
Unfortunately I don't know of any faithful implementations currently. Signatrix also doesn't faithfully implement EfficientDet as they aren't using the all stride2 version of EfficientNet that the original EfficientDet paper offhandly implies is correct (which admittedly this repo does get right). Unfortunately, everyone is up a creek without a paddle until someone either thoroughly vets and corrects problems with these codes (and one of the owners of the highly rated but flawed repos actually implements/merges the corrections) or the original authors release their code so it can be faithfully ported to PyTorch (since it was presumably in Tensorflow)

**Comment by macwilam** [comment]
Thanks a lot for the response. It is strange that there are no more replication attempts from other projects considering the claims regarding the efficiency and first results (some people are getting decent results from D0 if I understand correctly).

Well nothing else to do than to keep the fingers crossed.

**Comment by rmcavoy** [comment]
I am guessing that if we get a good PyTorch implementation it will come from Facebook making one to put in Detection2.
